.travis.yml

```diff
@@ -0,0 +1,65 @@
+language: android
+jdk: oraclejdk8
+
+sudo: false
+
+env:
+  global:
+    - NDK_VERSION=r11c
+    - BORINGSSL_HOME="$HOME/boringssl"
+    - CC=clang
+    - CXX=clang++
+
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/dists/
+    - $HOME/.cache/ndk
+before_cache:
+  - find $HOME/.gradle -name "*.lock" -exec rm {} \;
+  - rm -rf $HOME/.gradle/caches/[1-9]*
+
+before_script:
+  # newest Android NDK
+  - mkdir -p $HOME/.cache/ndk
+  - curl https://dl.google.com/android/repository/android-ndk-${NDK_VERSION}-linux-x86_64.zip -z $HOME/.cache/ndk/ndk-${NDK_VERSION}.zip -o $HOME/.cache/ndk/ndk-${NDK_VERSION}.zip
+  - unzip -q $HOME/.cache/ndk/ndk-$NDK_VERSION.zip -d $HOME
+  - echo "ndk.dir=$HOME/android-ndk-$NDK_VERSION" >> local.properties
+  # get and build BoringSSL
+  - mkdir $BORINGSSL_HOME
+  - git clone --depth 1 https://boringssl.googlesource.com/boringssl $BORINGSSL_HOME
+  - mkdir $BORINGSSL_HOME/build && pushd $BORINGSSL_HOME/build
+  - cmake -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DCMAKE_BUILD_TYPE=Release -DCMAKE_ASM_FLAGS=-Wa,--noexecstack -GNinja ..
+  - ninja
+  - popd
+
+before_install:
+  - echo 'android.builder.sdkDownload=false' > gradle.properties
+
+addons:
+  apt:
+    sources:
+      - kalakris-cmake
+      - ubuntu-toolchain-r-test
+    packages:
+      - cmake
+      - ninja-build
+      - clang
+
+android:
+  components:
+    - tools
+    - tools # https://github.com/travis-ci/travis-ci/issues/6040
+    - platform-tools
+    - extra-android-m2repository
+    - build-tools-25.0.0
```

Comment:
I don't think you can have substitutions here unfortunately. It would be nice to have a gradle rule that made sure they were in sync, though.